components/calendar/src/chinese_based.rs

```diff
@@ -0,0 +1,346 @@
+        adjusted_rem_euclid(3 + quotient(s - 15, 30), 12)
+    }
+
+    /// The fixed date in standard time at the observation location of the next new moon on or after a given Moment.
+    ///
+    /// Based on functions from _Calendrical Calculations_ by Reingold & Dershowitz.
+    /// Lisp reference code: https://github.com/EdReingold/calendar-code2/blob/main/calendar.l#L5329-L5338
+    pub(crate) fn new_moon_on_or_after(moment: Moment) -> RataDie {
+        let new_moon_moment = Astronomical::new_moon_at_or_after(Self::midnight(moment));
+        let location = C::location(new_moon_moment.as_rata_die());
+        Location::standard_from_universal(new_moon_moment, location).as_rata_die()
+    }
+
+    /// The fixed date in standard time at the observation location of the previous new moon before a given Moment.
+    ///
+    /// Based on functions from _Calendrical Calculations_ by Reingold & Dershowitz.
+    /// Lisp reference code: https://github.com/EdReingold/calendar-code2/blob/main/calendar.l#L5318-L5327
+    pub(crate) fn new_moon_before(moment: Moment) -> RataDie {
+        let new_moon_moment = Astronomical::new_moon_before(Self::midnight(moment));
+        let location = C::location(new_moon_moment.as_rata_die());
+        Location::standard_from_universal(new_moon_moment, location).as_rata_die()
+    }
+
+    /// Universal time of midnight at start of a Moment's day at the observation location
+    ///
+    /// Based on functions from _Calendrical Calculations_ by Reingold & Dershowitz.
+    /// Lisp reference code: https://github.com/EdReingold/calendar-code2/blob/main/calendar.l#L5353-L5357
+    pub(crate) fn midnight(date: Moment) -> Moment {
+        Location::universal_from_standard(date, C::location(date.as_rata_die()))
+    }
+
+    /// Determines the fixed date of the lunar new year in the sui4 (solar year based on the winter solstice)
+    /// which contains the fixed date passed as an argument.
+    ///
+    /// Based on functions from _Calendrical Calculations_ by Reingold & Dershowitz.
+    /// Lisp reference code: https://github.com/EdReingold/calendar-code2/blob/main/calendar.l#L5370-L5394
+    pub(crate) fn new_year_in_sui(date: RataDie) -> RataDie {
+        let prior_solstice = Self::winter_solstice_on_or_before(date); // s1
+        let following_solstice = Self::winter_solstice_on_or_before(prior_solstice + 370); // s2
+        let month_after_eleventh = Self::new_moon_on_or_after((prior_solstice + 1).as_moment()); // m12
+        let month_after_twelfth =
+            Self::new_moon_on_or_after((month_after_eleventh + 1).as_moment()); // m13
+        let next_eleventh_month = Self::new_moon_before((following_solstice + 1).as_moment()); // next-m11
+        let m12_float = month_after_eleventh.as_moment().inner();
+        let next_m11_float = next_eleventh_month.as_moment().inner();
+        let lhs_argument = libm::round((next_m11_float - m12_float) / MEAN_SYNODIC_MONTH) as i64;
```

Comment:
Suggestion 1: just remove the `.inner()` since `impl Sub for Moment` returns `f64`

Suggestion 2: do the subtraction in RataDie space and cast to f64 later:

```
let lhs_argument = libm::round((month_after_eleventh - next_eleventh_month) as f64 / MEAN_SYNODIC_MONTH) as i64;
```

Reply:
I've went with the second suggestion; since `m12_float` and `next_m11_float` aren't needed anywhere except here, I've deleted them